### ISSUE TITLE
Repeater configurations broken

### ISSUE THREAD

**Issue by lantis1008** [ericpaulbishop/gargoyle#471]
In the latest code the repeater configs seem broken. 
I think that may be my fault in the AC rewrite. 
I'll take it upon myself to fix. 

It's a good opportunity to also fix up interaction with non standard radio numbering implementations e.g. Archer

**Comment by lantis1008** [comment]
Do I not understand it correctly, or does the ARPNAT patch require updating to suit 3.18.23 kernel?

**Comment by ericpaulbishop** [comment]
Arpnat should already be ported to the latest kernel. I remember testing it
with the current kernel and seeing that it was working, and I also remember
having to update the hotplug script. I'm not sure what might be broken, but
it is worth looking into.

**Comment by lantis1008** [comment]
in the process of fixing the GUI.

getting some bizarre repeater issues in my testing. Not caused by bad configurations. 
Anything connected to the repeater can't get a DHCP address.
The following is a log from the main router (i.e. the signal to be repeated).
I'm assuming that the declines are occuring due to some kind of timeout? So we are losing our DHCP requests/ACKs?
This is beyond my expertise.
```
Sat Mar  5 21:00:16 2016 daemon.info dnsmasq-dhcp[26684]: DHCPDISCOVER(br-lan) 1c:ab:a7:ed:9f:16
Sat Mar  5 21:00:16 2016 daemon.info dnsmasq-dhcp[26684]: DHCPOFFER(br-lan) 192.168.1.208 1c:ab:a7:ed:9f:16
Sat Mar  5 21:00:20 2016 daemon.info dnsmasq-dhcp[26684]: DHCPREQUEST(br-lan) 192.168.1.208 1c:ab:a7:ed:9f:16
Sat Mar  5 21:00:20 2016 daemon.info dnsmasq-dhcp[26684]: DHCPACK(br-lan) 192.168.1.208 1c:ab:a7:ed:9f:16 iPad
Sat Mar  5 21:00:25 2016 daemon.info dnsmasq-dhcp[26684]: DHCPDECLINE(br-lan) 192.168.1.208 1c:ab:a7:ed:9f:16
Sat Mar  5 21:00:38 2016 daemon.info dnsmasq-dhcp[26684]: DHCPDISCOVER(br-lan) 1c:ab:a7:ed:9f:16
Sat Mar  5 21:00:38 2016 daemon.info dnsmasq-dhcp[26684]: DHCPOFFER(br-lan) 192.168.1.209 1c:ab:a7:ed:9f:16
Sat Mar  5 21:00:38 2016 daemon.info dnsmasq-dhcp[26684]: DHCPDISCOVER(br-lan) 1c:ab:a7:ed:9f:16
Sat Mar  5 21:00:38 2016 daemon.info dnsmasq-dhcp[26684]: DHCPOFFER(br-lan) 192.168.1.209 1c:ab:a7:ed:9f:16
Sat Mar  5 21:00:39 2016 daemon.info dnsmasq-dhcp[26684]: DHCPREQUEST(br-lan) 192.168.1.209 1c:ab:a7:ed:9f:16
Sat Mar  5 21:00:39 2016 daemon.info dnsmasq-dhcp[26684]: DHCPACK(br-lan) 192.168.1.209 1c:ab:a7:ed:9f:16 iPad
Sat Mar  5 21:00:44 2016 daemon.info dnsmasq-dhcp[26684]: DHCPDECLINE(br-lan) 192.168.1.209 1c:ab:a7:ed:9f:16
Sat Mar  5 21:00:58 2016 daemon.info dnsmasq-dhcp[26684]: DHCPDISCOVER(br-lan) 1c:ab:a7:ed:9f:16
Sat Mar  5 21:00:58 2016 daemon.info dnsmasq-dhcp[26684]: DHCPOFFER(br-lan) 192.168.1.210 1c:ab:a7:ed:9f:16
Sat Mar  5 21:00:58 2016 daemon.info dnsmasq-dhcp[26684]: DHCPDISCOVER(br-lan) 1c:ab:a7:ed:9f:16
Sat Mar  5 21:00:58 2016 daemon.info dnsmasq-dhcp[26684]: DHCPOFFER(br-lan) 192.168.1.210 1c:ab:a7:ed:9f:16
Sat Mar  5 21:00:59 2016 daemon.info dnsmasq-dhcp[26684]: DHCPREQUEST(br-lan) 192.168.1.210 1c:ab:a7:ed:9f:16
Sat Mar  5 21:00:59 2016 daemon.info dnsmasq-dhcp[26684]: DHCPACK(br-lan) 192.168.1.210 1c:ab:a7:ed:9f:16 iPad
Sat Mar  5 21:01:14 2016 daemon.info dnsmasq-dhcp[26684]: DHCPDISCOVER(br-lan) 1c:ab:a7:ed:9f:16
Sat Mar  5 21:01:14 2016 daemon.info dnsmasq-dhcp[26684]: DHCPOFFER(br-lan) 192.168.1.210 1c:ab:a7:ed:9f:16
Sat Mar  5 21:01:15 2016 daemon.info dnsmasq-dhcp[26684]: DHCPREQUEST(br-lan) 192.168.1.210 1c:ab:a7:ed:9f:16
Sat Mar  5 21:01:15 2016 daemon.info dnsmasq-dhcp[26684]: DHCPACK(br-lan) 192.168.1.210 1c:ab:a7:ed:9f:16 iPad
Sat Mar  5 21:01:20 2016 daemon.info dnsmasq-dhcp[26684]: DHCPDECLINE(br-lan) 192.168.1.210 1c:ab:a7:ed:9f:16
```

**Comment by lantis1008** [comment]
Are you open to the use of relayd instead of ebtables? This would replicate the ARPNAT functionality but the relayd package is ~10kB and ebtables is ~60kB.

**Comment by ericpaulbishop** [comment]
Yes! I hadn't realized that existed. That would be better. We can't get rid of ebtables, but we can get rid the arpnat module if that works. ebtables is still necessary for identifying the guest network.

Also, so long as we're ditching 4M routers, space isn't quite as big of a deal.  In terms of space... the Rubicon/point of no return there is the known-devices branch as that relies on ipset. With ipset included (which used to be ok, but is now an absolute no-go on 4M devices), there's no turning back to the 4M routers.

**Comment by lantis1008** [comment]
Cool. I'll work on that then. This blows out this fix to around next weekend in that case. I'll do the GUI and the backend at the same time.

Did not realise guests still rely on ebtables! 
What i might do (just so i don't break anything) is implement the relayd functionality, and then let you decipher what needs removing. I'll add suggested removals where i'm confident so you don't have to dig too hard.

**Comment by lantis1008** [comment]
I think i've fixed everything. I'm now going to test for devices that are not N capable and devices with only single interfaces.
Progress so far: https://www.diffchecker.com/evelr6hv

I've also written a hotplug to bring up the bridge when required.

**Comment by lantis1008** [comment]
Patch incoming within 24 hours.

**Comment by lantis1008** [comment]
Resolved by 1f4726222fb9a0c1f2b8a6227ae8cedf73ca0d98
